api/controllers/ApplicationTypeController.js

```diff
@@ -54,93 +54,90 @@ module.exports = {
+        }
+
+        return res.view('applicationForms/applicationType.ejs', {
+            ...pageData,
+            back_link: '/',
+        });
     },
 
-    handleServiceChoice(req, res) {
-        const {'choose-a-service': chosenService} = req.body;
-        const userLoggedIn = HelperService.LoggedInStatus(req);
+    async _addUserAccountToSession({req, res, userModels, pageData}) {
+        try {
+            const userDataFromDB = await userModels.User.findOne({where: {email: req.session.email}});
+            const accountDataFromDB = await userModels.AccountDetails.findOne({where: {user_id: userDataFromDB.id}});
+            const standardAppCountQuery = 'SELECT count(*) FROM "Application" WHERE "user_id" =:userId and "serviceType" = 1 and "createdAt" > NOW() - INTERVAL \'' + sails.config.standardServiceRestrictions.appSubmissionTimeFrameInDays + ' days\' and ("submitted" =:submitted OR "submitted" =:queued)';
+
+            await sequelize.query(
+                standardAppCountQuery,
+                {
+                    replacements: {
+                        userId: userDataFromDB.id,
+                        submitted: 'submitted',
+                        queued: 'queued'
+                    },
+                type: sequelize.QueryTypes.SELECT
+            });
+
+            req.session.user = userDataFromDB;
+            req.session.account = accountDataFromDB;
+            req.session.email_sent = false;
 
-        if(!userLoggedIn) {
-            sails.log.error('User is not logged in');
-            return res.view('404');
+            return res.view('applicationForms/applicationType.ejs', {
+                ...pageData,
+                back_link: false,
+            });
+        } catch (err) {
+            sails.log.error(err);
+            res.serverError(err);
         }
+    },
 
+    handleServiceChoice(req, res) {
+        const chosenService = req.body['choose-a-service'];
         const servicePages = {
             eApostille: '/new-application?app_type_group=4',
             standard: '/new-application?app_type_group=1',
             premium: '/new-application?app_type_group=2',
             dropoff: '/new-application?app_type_group=3',
+            default: '/select-service',
         };
 
         if (!chosenService) {
             sails.log.error('No service selected');
-            return res.view('eApostilles/applicationType.ejs', {
-                userServiceURL: req._sails.config.customURLs.userServiceURL,
-                error_report: true,
-                user_data: HelperService.getUserData(req, res),
-                back_link: req.session.startBackLink,
-            });
+            req.flash('serviceSelectError', 'You must select a service type.');
+
+            return res.redirect('/select-service');
         }
-        return res.redirect(servicePages[chosenService]);
+        return res.redirect(servicePages[chosenService] || servicePages.default);
```

Comment:
I'm not sure if it will. `servicePages ` is an object with 5 keys and one of them is `default`. 

If none of the keys values are matched, `servicePages ` will just return `undefined`. 

<img width="276" alt="Screenshot 2022-04-04 at 17 37 34" src="https://user-images.githubusercontent.com/1377253/161590937-a1c07a20-ddd5-486c-b634-d5ac5095e2fc.png">

So the `|| servicePages.default` is supposed to prevent the site from breaking on the user.

If you're referring to the at line **134** `if (!chosenService) {` should catch scenarios where there are no matches, then yes that's a fair point.

Reply:
Got it.  Thanks for clarifying